pkgs/misc/emulators/rpcs3/default.nix

```diff
@@ -8,30 +8,54 @@
 }:
 
 let
-  majorVersion = "0.0.19";
-  gitVersion = "12975-37383f421";
+  # Keep these separate so the update script can regex them
+  rpcs3GitVersion = "13222-8c2fd5095";
+  rpcs3Version = "0.0.20-13222-8c2fd5095";
+  rpcs3Revision = "8c2fd50957be3af05c04a9bb782dce8505fb6400";
+  rpcs3Sha256 = "1cf62vpqdc9i4masgv9zz24h7zdc7gcymx6n1hbh7wp5gg1dw4qi";
+
+  ittapi = fetchFromGitHub {
+    owner = "intel";
+    repo = "ittapi";
+    rev = "v3.18.12";
+    sha256 = "0c3g30rj1y8fbd2q4kwlpg1jdy02z4w5ryhj3yr9051pdnf4kndz";
+  };
 in
 gcc11Stdenv.mkDerivation {
   pname = "rpcs3";
-  version = "${majorVersion}-${gitVersion}";
+  version = rpcs3Version;
 
   src = fetchFromGitHub {
     owner = "RPCS3";
     repo = "rpcs3";
-    rev = "37383f4217e1c510a543e100d0ca495800b3361a";
+    rev = rpcs3Revision;
     fetchSubmodules = true;
-    sha256 = "1pm1r4j4cdcmr8xmslyv2n6iwcjldnr396by4r6lgf4mdlnwahhm";
+    sha256 = rpcs3Sha256;
   };
 
   passthru.updateScript = ./update.sh;
 
   preConfigure = ''
     cat > ./rpcs3/git-version.h <<EOF
-    #define RPCS3_GIT_VERSION "${gitVersion}"
+    #define RPCS3_GIT_VERSION "${rpcs3GitVersion}"
     #define RPCS3_GIT_FULL_BRANCH "RPCS3/rpcs3/master"
     #define RPCS3_GIT_BRANCH "HEAD"
     #define RPCS3_GIT_VERSION_NO_UPDATE 1
     EOF
+
+    # Hack around LLVMs ittapi dodgyness
```

Comment:
Can we just patch that out instead?

Reply:
Patching it out would require modifying LLVM's build system, and I'm not really game to mess with that.

As far as hacks go, it's pretty clean. I'm just working around the fact that it can't `git clone` from a remote repo inside the build sandbox.

rpcs3 also uses a custom LLVM fork in a submodule, which makes it even more annoying.

This is the relevant CMake code:
```cmake
if(NOT DEFINED ITTAPI_GIT_REPOSITORY)
    set(ITTAPI_GIT_REPOSITORY https://github.com/intel/ittapi.git)
endif()

if(NOT DEFINED ITTAPI_GIT_TAG)
    set(ITTAPI_GIT_TAG v3.18.12)
endif()

if(NOT DEFINED ITTAPI_SOURCE_DIR)
    set(ITTAPI_SOURCE_DIR ${PROJECT_BINARY_DIR})
endif()

if(NOT EXISTS ${ITTAPI_SOURCE_DIR}/ittapi)
    execute_process(COMMAND ${GIT_EXECUTABLE} clone ${ITTAPI_GIT_REPOSITORY}
                    WORKING_DIRECTORY ${ITTAPI_SOURCE_DIR}
                    RESULT_VARIABLE GIT_CLONE_RESULT)
    if(NOT GIT_CLONE_RESULT EQUAL "0")
        message(FATAL_ERROR "git clone ${ITTAPI_GIT_REPOSITORY} failed with ${GIT_CLONE_RESULT}, please clone ${ITTAPI_GIT_REPOSITORY}")
    endif()
endif()

execute_process(COMMAND ${GIT_EXECUTABLE} checkout ${ITTAPI_GIT_TAG}
                WORKING_DIRECTORY ${ITTAPI_SOURCE_DIR}/ittapi
                RESULT_VARIABLE GIT_CHECKOUT_RESULT)
if(NOT GIT_CHECKOUT_RESULT EQUAL "0")
    message(FATAL_ERROR "git checkout ${ITTAPI_GIT_TAG} failed with ${GIT_CHECKOUT_RESULT}, please checkout ${ITTAPI_GIT_TAG} at ${ITTAPI_SOURCE_DIR}/ittapi")
endif()

include_directories( ${ITTAPI_SOURCE_DIR}/ittapi/include/ 
```

Reply:
Can we just drop execute_process?

Reply:
Done, I've just removed the offending code